ci/build-project.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -e -u -o pipefail
+if [[ -n "${DEBUG-}" ]]; then
+    set -x
+fi
+
+pushd "$(dirname "$0")"
+    UNREAL_PATH="${1?Please enter the path to the Unreal Engine.}"
+    TEST_REPO_BRANCH="${2?Please enter the branch that you want to test.}"
+    TEST_REPO_URL="${3?Please enter the URL for the git repo you want to test.}"
+    TEST_REPO_UPROJECT_PATH="${4?Please enter the path to the uproject inside the test repo.}"
+    TEST_REPO_PATH="${5?Please enter the path where the test repo should be cloned to.}"
+    GDK_HOME="${6?Please enter the path to the GDK for Unreal repo.}"
+    BUILD_PLATFORM="${7?Please enter the build platform for your Unreal build.}"
+    BUILD_STATE="${8?Please enter the build state for your Unreal build.}"
+    BUILD_TARGET="${9? Please enter the build target for your Unreal build.}"
+
+    # Clone the testing project
+    rm -rf "${TEST_REPO_PATH}"
+    echo "Downloading the testing project from ${TEST_REPO_URL}"
```

Comment:
```suggestion
    echo "Cloning the testing project from ${TEST_REPO_URL}"
```

Reply:
Should this also have the option of a `TEST_PROJECT_BRANCH` which we checkout to once cloned?

Reply:
Could use `--branch {branch} --single-branch` to speed up that clone and be on the branch you want.